Apps/Listener/dev-commands.lisp

```diff
@@ -234,20 +234,29 @@
 ;; McCLIM fixme: Shouldn't we be able to activate before the (args) prompt
 ;; since defaults are defined?
 ;; FIXME: Disabled input, as it usually seems to hang.
-(define-command (com-run :name "Run" :command-table application-commands :menu t
-			 :provide-output-destination-keyword t)
+(define-command (com-run :name "Run" :command-table application-commands :menu t)
   ((program 'string :prompt "Command")
-   (args '(sequence string) :default nil :prompt "Arguments"))
-  (run-program program args :wait t :input nil))
+   (args '(sequence string) :default '("") :prompt "Arguments"))
+  (let ((output-stream *standard-output*))
+    (with-text-family (output-stream :fix)
+      (if (zerop (length (car args)))
+	  (progn (heading "Runnig \"~A\"~%" program)
+		 (uiop:run-program program :force-shell nil :output output-stream :input nil
+					   :ignore-error-status t :error-output output-stream))
+	  (progn (heading "Running \"~A ~{~A ~}\"~%" program args)
+		 (uiop:run-program `(,program ,@args) :force-shell nil :output output-stream
+						      :input nil :ignore-error-status t
+						      :error-output output-stream))))))
 
 ;; I could replace this command with a keyword to COM-RUN..
 (define-command (com-background-run :name "Background Run"
                                     :menu t
-				    :command-table application-commands
-				    :provide-output-destination-keyword t)
+				    :command-table application-commands)
   ((program 'string :prompt "Command")
    (args '(sequence string) :default nil :prompt "Args"))
-  (run-program program args :wait nil :output nil :input nil))
+  (if (zerop (length (car args)))
+      (uiop:launch-program program)
+      (uiop:launch-program `(,program ,@args))))
```

Comment:
I mean here: this is launch-program not run-program.

Reply:
ah, I take that back, run-program in uiop doesn't cover `wait` argument.

Reply:
Oh, this is in com-background-run command, it is meant to run asynchronously.